lib/absinthe/subscription.ex

```diff
@@ -106,14 +106,15 @@ defmodule Absinthe.Subscription do
   defp fetch_fields(_, _), do: []
 
   @doc false
-  def subscribe(pubsub, field_key, doc_id, doc) do
+  def subscribe(pubsub, field_key, doc_id, doc, options) do
     registry = pubsub |> registry_name
 
     doc_value = {
       doc_id,
       %{
-        initial_phases: doc.initial_phases,
```

Comment:
The more I think about this the more I'm certain that this is not an equivalent change. `doc.initial_phases` is going to capture any and all changes that the end user made to the pipeline that was run, but when we throw that away and instead use ` pipeline = Absinthe.Pipeline.for_document(doc.schema, doc.options)` there is no real reason to believe it will return the same pipeline. All that `for_document` does is return the basic default pipeline, it won't re-run any pipeline modifiers that the end user has.

Reply:
Ok, then indeed I might have misunderstood the flow here 😓 . I went off the assumption that `initial_phases` is usually populated at the very beginning (by `Init` phase) - and that `SubscribeSelf` uses that unchanged, initial pipeline instead of the actual flow adjusted by user. This does not hold though, right?

If that's the case, then perhaps, indeed, a "compression" approach, where occurrences of unique options kwlists could be stored in a separate map and the pipeline would hold only their placeholders, (`{:options, 1}`, `{:options, 2}` etc.). The pipeline could be easily reconstructed then when publishing it.